src/plone/exportimport/utils/principals/members.py

```diff
@@ -78,8 +78,8 @@ def _get_base_user_data(member: MemberData):
     # username, groups, roles
     props = {
         "username": user_id,
-        "roles": json_compatible(roles),
-        "groups": json_compatible(groups),
+        "roles": json_compatible(sorted(roles)),
+        "groups": json_compatible(sorted(groups)),
```

Comment:
I would do the sorting in helpers.get_all_groups

Reply:
I did not do that initially, because get_all_groups returns a list of `GroupData` items, and in here we have a simple list of group ids.  But sure, let's keep the helpers the same and let all of them return a sorted list.  Done.